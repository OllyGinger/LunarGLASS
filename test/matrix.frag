#version 130

uniform mat3 colorTransform;
varying vec3 Color;
uniform mat4 m, n;
uniform mat4x3 um43;
uniform mat3x4 un34;
varying vec4 v;
varying vec3 u;

void main()
{
    gl_FragColor = vec4(0.0);
    gl_FragColor += vec4(Color * colorTransform, 1.0);

    if (m != n)
        gl_FragColor += v;
    else {
        gl_FragColor += m * v;
        gl_FragColor += v * (m - n);
    }
    
    mat3x4 m34 = outerProduct(v, u);
    if (m34 == un34)
        gl_FragColor += m34 * u;
    else
        gl_FragColor += (un34 * um43) * v;
}
