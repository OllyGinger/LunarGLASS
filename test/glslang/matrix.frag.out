#### BEGIN COMPILER 0 INFO LOG ####
ERROR: 0:23: 'outerProduct' : no matching overloaded function found 
ERROR: 0:23: '=' :  cannot convert from 'const float' to '4X4 matrix of float'
ERROR: 0:25: '*' :  wrong operand types  no operation '*' exists that takes a left-hand operand of type '4X4 matrix of float' and a right operand of type 'varying in 3-component vector of float' (or there is no acceptable conversion)
ERROR: 0:25: 'assign' :  cannot convert from '4X4 matrix of float' to 'FragColor 4-component vector of float'
ERROR: 4 compilation errors.  No code generated.


#### END COMPILER 0 INFO LOG ####
#### BEGIN LINKER INFO LOG ####
ERROR: Not all shaders have valid object code.

#### END LINKER INFO LOG ####
